src/main/scala/io/qbeast/spark/table/IndexedTable.scala

```diff
@@ -513,34 +533,65 @@ private[table] class IndexedTableImpl(
     }
   }
 
-  override def optimize(revisionID: RevisionID, options: Map[String, String]): Unit =
-    optimize(revisionID, 1.0, options)
+  override def optimizeUnindexedFiles(
+      unindexedFiles: Seq[String],
+      options: Map[String, String]): Unit = {
+    if (unindexedFiles.isEmpty) return // Nothing to optimize
+    // 1. Load the files from the Staging ID (Unindexed)
+    val files =
+      snapshot.loadIndexFiles(stagingID).filter(f => unindexedFiles.contains(f.path))
+    import files.sparkSession.implicits._
+    // 2. Load the Dataframe, the index status and the schema
+    val filesDF = snapshot.loadDataframeFromIndexFiles(files)
+    val indexStatus = snapshot.loadIndexStatus(stagingID)
```

Comment:
This `IndexStatus` will be used to index that data, and it has to be that of the latest revision.

Reply:
thank god you review it.. I will add tests as well